### ISSUE TITLE
BC: error when selecting or trailing a bubble that is not on screen

### ISSUE THREAD

**Issue by angieskazka** [Gapminder/tools-page#190]
Start bubble chart
Change Y to literacy rate, total
select Mali in the select list
select China in the select list (striken through)
`Cannot read property 'label' of undefined at __labelWithoutFrame()`

China bubble then apprars out of nowhere

expected behavior: move time to the nearest point where china is available and select china

Make trails for Mali and China, so that chart stops at year 2011
Refresh
see more errors
`Cannot read property 'getByObjOrStr' of undefined at trail.js:83`

here is a link to the final state

http://www.localhost:4200/tools/#$model$markers$bubble$encoding$selected$data$filter$markers$country-mli:true&country-chn:true;;;;&y$data$concept=literacy_rate_adult_total_percent_of_people_ages_15_and_above&space@=country&=time;;&scale$domain:null&zoomed:null;;&trail$starts$country-mli=1980-08-04T04:18:16.593Z&country-chn=2005-11-17T02:24:17.314Z;&data$filter$markers$country-mli:true&country-chn:true;;;;;;;;&chart-type=bubbles&url=v1

**Comment by jheeffer** [comment]
You're asking for chinese marker from a frame that doesn't contain a chinese marker...

`const trailData = this.model.getDataMapByFrameValue(trailStart).getByObjOrStr(null, key);`

because you set trail start to a non existing bubble
```
          this.MDL.trail.setTrail({
            [Symbol.for("key")]: marker,
            [this.MDL.trail.groupDim]: this.MDL.frame.value
          });
```
So this is expected.

I updated vizabi `trail.setTrail` to clamp to limits to prevent faulty sets like these to still end up as valid config.

Furthermore, frame is completely stripped from url. So when refreshing, it tries to display 2019, and especially with splashmarker, which has no other frames to fall back to, that of course crashes. So I put `frame` back into url.

Updated some other parts of components that expected each selected bubble to always be in frame.

Question: When selecting crossed out China, why would time jump to 2010 (when china is available) and not just do what it does now: Set trailstart to nearest available time (2010) and thus china bubble for 2010 shows up.
